components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSimplePAPStatusDataHandler.java

```diff
@@ -0,0 +1,389 @@
+                } finally {
+                    IdentityDatabaseUtil.closeConnection(connection);
+                }
+            }
+        }
+    }
+
+    private List<StatusHolder> getStatus(String key, String about) throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        List<StatusHolder> statusHolders = new ArrayList<>();
+        String query = EntitlementConstants.Status.ABOUT_POLICY.equals(about)
+                ? GET_POLICY_STATUS_SQL
+                : GET_SUBSCRIBER_STATUS_SQL;
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (NamedPreparedStatement getStatusPrepStmt = new NamedPreparedStatement(connection, query)) {
+                getStatusPrepStmt.setString(KEY, key);
+                getStatusPrepStmt.setInt(TENANT_ID, tenantId);
+                try (ResultSet statusSet = getStatusPrepStmt.executeQuery()) {
+                    while (statusSet.next()) {
+                        StatusHolder statusHolder = new StatusHolder(about);
+                        if (EntitlementConstants.Status.ABOUT_POLICY.equals(about)) {
+                            statusHolder.setKey(statusSet.getString(POLICY_ID));
+                        } else {
+                            statusHolder.setKey(
+                                    statusSet.getString(SUBSCRIBER_ID));
+                        }
+                        statusHolder.setType(statusSet.getString(STATUS_TYPE));
+                        statusHolder.setSuccess(statusSet.getBoolean(IS_SUCCESS));
+                        statusHolder.setUser(statusSet.getString(USER));
+                        statusHolder.setTarget(statusSet.getString(TARGET));
+                        statusHolder.setTargetAction(statusSet.getString(TARGET_ACTION));
+                        statusHolder.setTimeInstance(String.valueOf(statusSet.getTimestamp(LOGGED_AT).getTime()));
+                        statusHolder.setMessage(statusSet.getString(MESSAGE));
+
+                        String version = null;
+                        if (EntitlementConstants.Status.ABOUT_POLICY.equals(about) &&
+                                statusSet.getInt(POLICY_VERSION) != -1) {
+                            version = Integer.toString(statusSet.getInt(POLICY_VERSION));
+                        }
+                        statusHolder.setVersion(version);
+                        statusHolders.add(statusHolder);
+                    }
+                }
+                return statusHolders;
+            }
+        } catch (SQLException e) {
+            throw new EntitlementException("Error while retrieving policy status", e);
+        }
+    }
+
+    private void addStatus(String about, String key, List<StatusHolder> statusHolders) throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
```

Comment:
Shall we pass tenantId as a param to keep consistency of other pure DAO methods?